File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
        List<Integer> list = new ArrayList<>();
        int length = 0;

        while (head != null) {
            list.add(head.val);
            head = head.next;
            length++;
        }

        int biggestSum = 0;

        for (int i = 0; i< length/2; i++){
            int sum = list.get(i) + list.get(length-i-1);
            biggestSum = sum > biggestSum? sum : biggestSum;
        }

        return biggestSum;
    }
}

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func pairSum(head *ListNode) int {
    list := make([]int, 0)
    var length int
    
    for {
        if head == nil {
            break
        }
        list = append(list, head.Val)
        head = head.Next
        length++
    }
    
    var maxSum int
    for i:=0; i< length/2; i++ {
        sum := list[i] + list[length-1-i]
        if sum > maxSum {
            maxSum = sum
        }
    }
    
    return maxSum
}